admin_guide/high_availability.adoc

```diff
@@ -242,6 +243,101 @@ The list of command options described here are a subset that are relevant to thi
 Each VIP in the set may end up being served by a different node.
 ====
 
+[[check-notify]]
+=== Check and Notify scripts
+
+The *keepalived* port monitoring feature uses a simple port check ( </dev/tcp/<ip>/<watch-port> ) to verify that the application is running. The admin can provide a script that does whatever additional verification is needed, for example, the script can test a web server by issuing a request and verifying the response. The script must exit with 0 for PASS and 1 for FAIL.
+
+The admin provides the additional script, via the `--check-script=<script>` option.  By default the check is done every 2 seconds, this can be changed using the `--check-interval=<seconds>` option.
+
+For each VIP, *keepalived* keeps the state of the node. The VIP on the node may be in *MASTER*, *BACKUP*, or *FAULT* state.  All nodes that are not in the *FAULT* state negotiate to decide who will be *MASTER* for the VIP.  All of the losers enter the *BACKUP* state.  When the check script fails *keepalived* enters the *FAULT* state. When the check script passes again it exits *FAULT* and negotiates for *MASTER*. The resulting state is either *MASTER* or *BACKUP*.
```

Comment:
The every keepalived (on all nodes) runs the check script on a set period.  When MASTER fails, trigger a renegotiation for MASTER. When BACKUP fails just enter FAULT. When FAULT fails do nothing. When FAULT passes, trigger a renegotiation. When MASTER or BACKUP passes, do nothing.